### DISCUSSION TITLE
Outreachy internship: Make enhancements to user profile

### DISCUSSION THREAD

**Discussion by Deepanshu039** [unstructuredstudio/zubhub#498] [Ideas]
These are the categories I was thinking of in which we can assign badges to the creators: -

On the basis of Project count ---> (1+ project, 10+ project, 100+project )
On the basis of total views count ---> (1k+ views, 10k+ views, 100k+views)
On the basis of the Likes count ---> ( 100+ likes, 1k+ likes, 10+ likes)
On the basis of Comments a creator makes ---> (100+ comments, 500+ comments, 1k+ comments)
[optional] On the basis of followers ---> (100+ followers, 1k+ followers , 10k+ followers)
since these are just the categories in which we will assign badges to the creators the actual title which will display on the user profiles will be different.
The appropriate title can be discussed and can be assigned to the particular count, also the no of counts for a particular count can be increased and decreased as instructed.

I have prepared these two mockups for adding achievements components to the user profile and would really like to get your feedback on this @srish @tuxology .

![Adobe Scan 04-Jun-2022_1](https://user-images.githubusercontent.com/92849065/172202859-59aab9e2-bc74-4e7a-a928-dabd121f8958.jpg)

since @tuxology suggests initially we will consider tags for the achievements so I thought of this, later on, these tags will be displayed with the medals or trophies which will be associated with a specific tag.

**Comment by tuxology** [comment]
Comment from @srish about above stuff here: https://github.com/unstructuredstudio/zubhub/issues/484#issuecomment-1148014680 Lets continue discussions related to this here :+1:

**Comment by Deepanshu039** [comment]
@Deepanshu039 The initial design looks great and makes a lot of sense! :-) I also like the placement of achievement box. Sharing a few thoughts:

First on what additional badges to consider besides the one you have already shared:

On the basis of comments a creator has received
Project featured by studio / Featured on homepage. For example, see "Projects from our recent creative learning workshops" on the homepage
Custom badges that studio should be able to award from the backend
I think this might be too ambitious for now, but in the future, based on the category someone shares a lot of projects under (let's say "Art," then they can be awarded a badge of an "Artist")
Maybe in the achievements box, we can consider showing not more than 10 badges. And, if there are more badges, then they can be viewed on a separate page.

Other than these badges, I think we can consider showing a few more statistics:

Activity log and ability to pick a featured project on ZubHub. See for reference:
https://scratch.mit.edu/users/SrishAkaTux/

Number of projects, views, date of joining the platform, number of comments received, location. See for reference:
https://www.instructables.com/member/Markus+Opitz/

Maybe based on this comment, think about what is doable in the three months time and propose a modified mockup that I can take a look at. Just a thought, the priority order could be - badges, statistics, activity log, featured project. We don't need to try to complete everything :)

**Reply by Deepanshu039** [reply]
so that we can continue our discussion here only as @tuxology already suggested.

**Reply by Deepanshu039** [reply]
> On the basis of comments a creator has received 

 we can add one more category to the number of comments a creator has got on their profile, but wouldn't that be redundant if two creators are having any discussion in the comment section, that would increase the no. of comments for a creator, just asking....



> Project featured by studio / Featured on homepage. For example, see "Projects from our recent creative learning workshops" on the homepage

I do not get this point very clearly. plzz can you elaborate it a little bit.


> Custom badges that studio should be able to award from the backend
> I think this might be too ambitious for now, but in the future, based on the category someone shares a lot of projects under (let's say "Art," then they can be awarded a badge of an "Artist")

yeah, this feature can be implementable since we ask the project category every time a creator adds any project, so simply we can add a function that keeps the count of the project category for a creator, and whenever the count exceeds the threshold the creator will get a badge for the category. @srish @tuxology

**Reply by Deepanshu039** [reply]
I have prepared this mock-up, hope this can be of any use please provide your opinion regarding this @srish @tuxology 

![mock_up2](https://user-images.githubusercontent.com/92849065/172349943-f533a569-460d-48eb-af32-2dfe1eadb841.jpeg)

**Comment by Deepanshu039** [comment]
thankyou so much @srish for appreciating the design, first of all, can you plz suggest which mockup we should further proceed with.

for not displaying more than 10+ badges we can simply add a "+more" button, I will soon upload a mockup for this for better clarity.

**Comment by srish** [comment]
@Deepanshu039 This looks great! Sharing replies to your questions below:


> we can add one more category to the number of comments a creator has got on their profile, but wouldn't that be redundant if two creators are having any discussion in the comment section, that would increase the no. of comments for a creator, just asking....
> 
> >

I think it is fine unless we are not counting the comments made by the creator themselves. 


> Project featured by studio / Featured on homepage. For example, see "Projects from our recent creative learning workshops" on the homepage

> > I do not get this point very clearly. plzz can you elaborate it a little bit.

I meant in case a creator's project is featured on the homepage (see "Projects from our recent creative learning workshops
" here https://zubhub.unstructured.studio/), then could they be awarded a badge for it?

Another idea that @tuxology brought was where children will be able to nominate others to receive a particular badge. But, I think in the first iteration, this might be too ambitious.

Other than this, I think your mockup content wise looks good. @tuxology may have some comments about the design and next steps. Also, we had a discussion around how much we should make the available badges public on the platform. And, if that might foster [extrinsic behaviors/ motivation](https://en.wikipedia.org/wiki/Motivation#Types_of_motivation) which we want to avoid. See a question around this in the meeting doc which we could ask our guest speaker in the upcoming session Friday.

**Reply by Deepanshu039** [reply]
> > we can add one more category to the number of comments a creator has got on their profile, but wouldn't that be redundant if two creators are having any discussion in the comment section, that would increase the no. of comments for a creator, just asking....
> > >
> 
> I think it is fine unless we are not counting the comments made by the creator themselves.

@srish  okay, this badge can work in this way. 


> I meant in case a creator's project is featured on the homepage (see "Projects from our recent creative learning workshops
> " here https://zubhub.unstructured.studio/), then could they be awarded a badge for it?

yeah, this is indeed an encouraging feature, right now we are displaying those videos in the sections which have the most views. So, we can simply give the creator a badge whose video has the most views and display it on the home page.

also @tuxology  I am awaiting your comments on the initial mock-up. Since I preparing a figma file for the same, so I can edit the suggestions you will make in the design.

**Comment by Deepanshu039** [comment]
Greetings @srish @tuxology , this is the figma implementation of the rough mockup. I would love to know your feedback on this 
https://www.figma.com/file/qwDHghGfIPy9740jLylPwH/figma-mockup_1.0?node-id=0%3A1

**Comment by srish** [comment]
I would put the activity log on the right so that the attention stays on the projects. Also in the activity log, we can have icons at the beginning just like in the example I shared. I think we are missing the other user statistics right now in the mockup that should go maybe along with "X Project Y Followers Z following.. ". Apart from this, I think it might be good to still refine the design of the badges in parallel. I am not good at design but wondering if badges can each be of different color, shape, size, etc. Have we looked at badges in other platforms?

Having said that, I think you could proceed w/ coding. Maybe the priority order could be badges > activity log > stats.  I feel like some of these designs can be improved later as well. @tuxology what you think?

**Reply by Deepanshu039** [reply]
> I would put the activity log on the right so that the attention stays on the projects.

noted.


> Also in the activity log, we can have icons at the beginning just like in the example I shared 

even I was feeling something was missing in activity log, will definitely add icons to it.



> I think we are missing the other user statistics right now in the mockup that should go maybe along with "X Project Y Followers Z following..

@srish  what parameters can we use to represent user statistics such as :- view per week, or per month or any other parameters which can provide more insights of users.

> Apart from this, I think it might be good to still refine the design of the badges in parallel

will definitely work on this in parallel and also be sharing badges design in future

**Reply by Deepanshu039** [reply]
> I would put the activity log on the right so that the attention stays on the projects. Also in the activity log, we can have icons at the beginning just like in the example I shared. I think we are missing the other user statistics right now in the mockup that should go maybe along with "X Project Y Followers Z following.. ". Apart from this, I think it might be good to still refine the design of the badges in parallel. I am not good at design but wondering if badges can each be of different color, shape, size, etc. Have we looked at badges in other platforms?
> 
> Having said that, I think you could proceed w/ coding. Maybe the priority order could be badges > activity log > stats. I feel like some of these designs can be improved later as well. @tuxology what you think?

@srish, added some of the modifications you instructed, also working on the user statistic thing(soon uploading a design).
https://www.figma.com/file/qwDHghGfIPy9740jLylPwH/figma-mockup_1.0?node-id=0%3A1
@tuxology awaiting your comments also on the design.

**Reply by tuxology** [reply]
@Deepanshu039 I've left a few comments on the mosckups you did above. Generally, a few suggestions and lets think about mobile views also.

**Reply by tuxology** [reply]
I'll also suggest you begin coding the page already in parallel if you feel comfortable - esp the badge system since it may have some backend components also. I am still contemplating - do we need a separate badge API or we can do this directly under the creator API?

**Comment by tuxology** [comment]
@Deepanshu039 @srish About project names, I've come up with a few in English. We should also have some in Hindi as well. You will need these to fill in those chips. We can use Font awesome icons for now and just keep some neutral colors for badges - or maybe Yellow with red borders so they look different than rest. Some examples I came up with are here: 

On the basis of Project count:

1+ project -> Hatchling  (We can use this icon : https://fontawesome.com/icons/egg?s=solid)
10+ project -> Flying Bird (https://fontawesome.com/icons/dove?s=solid)
50+ project -> Master of the Sky (https://fontawesome.com/icons/plane-up?s=solid)
100+ projects -> Expert Builder

On the basis of total views count:

10+ views -> Getting Famous
100+ views -> Person of Interest
5k+ views -> Popular Projects
100k+views -> Idea Factory

On the basis of the Likes count: 
100+ likes -> Interesting Projects
500+ likes -> Favourite Kid
1K+ likes -> Captain Project 

On the basis of Comments a creator makes:

100+ comments -> Helping Hand 
500+ comments -> Always Available
1k+ comments -> Expert Advisor

An important thing to make sure is that we can easily change the badges names when needed. In some cases the custom ZubHub hosting users may want to edit names as per their own requirements. So lets keep an option in backend to keep them editable.
